proposal: run Discourse on TiDB (#109)
pingcap/community: rfc/2019-11-28-discourse-on-tidb.md

# Proposal: Migrate Discourse's db from PostgreSQL to TiDB

- Author(s): EE-Team
- Last updated:  Nov 19, 2019
- Discussion at: https://github.com/pingcap-incubator/discourse/issues/

## Abstract

> The proposal aim at:

Reviewer: ```suggestion
The proposal aims at:
```

> 1. **Eat our own dog food**;

Reviewer: ```suggestion
1. **Eating our own dog food**;
```

> 2. Connect TiDB Community with Discourse Community and Ruby On Rails Community;

Reviewer: ```suggestion
2. Connecting TiDB Community with Discourse Community and Ruby On Rails Community;
```

## Background

> The TiDB User Group Website: [AskTUG](https://asktug.com) is running on Discourse, and Discourse is running on PostgreSQL, we want to migrate it to TiDB. Then we can frequently test TiDB before each release.

Reviewer: ```suggestion
The TiDB User Group Website: [AskTUG](https://asktug.com) is running on Discourse, and Discourse is running on PostgreSQL. We want to migrate it to TiDB. Then we can frequently test TiDB before each release.
```

## Proposal

We will make these changes:

> 1. Make Discourse's most sql statemennt run on MySQL;

Reviewer: ```suggestion
1. Make most SQL statements of Discourse run on MySQL;
```

> 2. Add new component elastic search instead of PostgreSQL's fulltext search;

Reviewer: ```suggestion
2. Add new elastic search  componentinstead of PostgreSQL's full-text search;
```

> 3. Migrate from MySQL to TiDB, fix potential compatible problems;

Reviewer: ```suggestion
3. Migrate from MySQL to TiDB, and fix potential compatibility problems;
```

## Rationale

> Translate sql in PostgreSQL to MySQL within Rails's ORM

Reviewer: ```suggestion
Translate SQL statements from PostgreSQL to MySQL within Rails's ORM
```
